beacon-chain/db/filesystem/iteration.go

```diff
@@ -0,0 +1,230 @@
+package filesystem
+
+import (
+	"fmt"
+	"io"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	fieldparams "github.com/prysmaticlabs/prysm/v5/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v5/encoding/bytesutil"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+)
+
+var errIdentFailure = errors.New("failed to determine blob metadata, ignoring all sub-path.")
+
+type identificationError struct {
+	err   error
+	path  string
+	ident blobIdent
+}
+
+func (ide *identificationError) Error() string {
+	return fmt.Sprintf("%s path=%s, err=%s", errIdentFailure.Error(), ide.path, ide.err.Error())
+}
+
+func (ide *identificationError) Unwrap() error {
+	return ide.err
+}
+
+func (*identificationError) Is(err error) bool {
+	return err == errIdentFailure
+}
```

Comment:
These two functions appear to be unused, are they here just to "complete" `identificationError`?

Reply:
Implementing `Is`, `As` and `Unwrap` on a custom error allows you to control how the stdlib [errors](https://pkg.go.dev/errors) package deals with the error:
- [errors.As](https://pkg.go.dev/errors#As) `An error type might provide an As method so it can be treated as if it were a different error type. `
- [errors.Is](https://pkg.go.dev/errors#Is) `An error type might provide an Is method so it can be treated as equivalent to an existing error.`
- [errors.Unwrap](https://pkg.go.dev/errors#Unwrap)`Unwrap returns the result of calling the Unwrap method on err, if err's type contains an Unwrap method returning error.`

`errors.Is` and `errors.As` are directly used in this PR. `Unwrap` is used under the hood by `errors.As` and `errors.Is`, eg within `errors.Is` in the case where custom `Is` returns false, the error will be further unwrapped, so for instance if some code wanted to check whether the error was of the wrapped error's type, the custom `Unwrap` is needed for that check to operate correctly (if `Is` is defined, `Unwrap` is only called when the custom `Is` returns false to continue recursively unwrapping and checking [[src](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/errors/wrap.go;l=53-78)]).